CRM/Event/Form/Registration/AdditionalParticipant.php

```diff
@@ -210,7 +210,7 @@ public function buildQuickForm() {
     //add buttons
     $js = NULL;
     if ($this->isLastParticipant(TRUE) && empty($this->_values['event']['is_monetary'])) {
-      $js = ['onclick' => "return submitOnce(this,'" . $this->_name . "','" . ts('Processing') . "');"];
+      $this->submitOnce = TRUE;
```

Comment:
I'm not sure that setting the value here in buildForm() actually works? It seems like you have to set it at the form level like in CRM/Event/Form/Registration/Confirm.php. In particular when I am at this form the `<form>` tag doesn't have the data-submit-once attribute set and running a debugger it doesn't trigger the submitOnce function. E.g. when I go through it for this event (as anonymous) http://core-14519-39tw3.test-1.civicrm.org:8001/civicrm/event/info?reset=1&id=7

Reply:
@demeritcowboy that additionalParticipant will only show if you are registering > 1 participant for the event and would only show on the 2nd and other n ones what the problem more likely is with the location of this https://github.com/civicrm/civicrm-core/pull/14519/files#diff-0e6b4e48225761b02cc269e604299d31R467 but it equally could apply here i'm not sure if doing the submit once on just non monetary events makes sense surely we would want it on all events right?

Reply:
Right for additionalParticipant it would only be on 2nd+, but it doesn't seem to get applied when you set the property in buildQuickForm. And yes it's the same issue for the other places where it's only set in buildQuickForm.

Regarding monetary vs nonmonetary the plan eventually I guess would to be to set it everywhere on all forms, just at the moment this PR is only touching some places.

Reply:
D'oh. I see it now. `is_monetary` means "Paid Event"=yes.